tunnistamo/middleware.py

```diff
@@ -26,6 +28,33 @@ def add_params_to_url(url, params):
     return url_parts._replace(query=urlencode(new_query_parts)).geturl()
 
 
+class LocaleMiddleware(DjangoLocaleMiddleware):
+    def process_request(self, request):
+        try:
+            language = request.GET.get('ui_locales')
+            language = get_supported_language_variant(language)
+            translation.activate(language)
```

Comment:
This needs more work. The `ui_locales` can be a space separated list of language codes. The first usable should be chosen. I will fix this.

Reply:
Thank you!

Reply:
Fixed with the new version.